kernel/userspace.c

```diff
@@ -0,0 +1,195 @@
+/*
+ * Copyright (c) 2017 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+
+#include <kernel.h>
+#include <string.h>
+#include <misc/printk.h>
+#include <kernel_structs.h>
+#include <sys_io.h>
+
+/**
+ * Kernel object validation function
+ *
+ * Retrieve metadata for a kernel object. This function is implemented in
+ * the gperf script footer, see gen_kobject_list.py
+ *
+ * @param obj Address of kernel object to get metadata
+ * @return Kernel object's metadata, or NULL if the parameter wasn't the
+ * memory address of a kernel object
+ */
+__weak _GENERIC_SECTION(.kobject_data.text)
+struct _k_object *_k_object_find(void *obj)
+{
+	/* This is the __weak stub implementation, the version created by
+	 * gperf will replace it in build phase 2
+	 */
+	ARG_UNUSED(obj);
+	return NULL;
+}
+
+#ifdef CONFIG_PRINTK
+const char *otype_to_str(enum k_objects otype)
+{
+	switch (otype) {
+	case K_OBJ_ALERT:
+		return "k_alert";
+	case K_OBJ_DELAYED_WORK:
+		return "k_delayed_work";
+	case K_OBJ_MEM_SLAB:
+		return "k_mem_slab";
+	case K_OBJ_MSGQ:
+		return "k_msgq";
+	case K_OBJ_MUTEX:
+		return "k_mutex";
+	case K_OBJ_PIPE:
+		return "k_pipe";
+	case K_OBJ_SEM:
+		return "k_sem";
+	case K_OBJ_STACK:
```

Comment:
Oh right, k_stack.  Remind me again why k_stack and k_fifo are different things (beyond the fact that one takes a void* and the other a u32_t)?  Not relevant to review, just something we should consider reviewing and unifying.

Reply:
yeah we have some issues with these APIs, additionally see https://jira.zephyrproject.org/browse/ZEP-2237